bundles/org.eclipse.swt/Eclipse SWT Tests/win32/org/eclipse/swt/internal/PlatformSpecificExecution.java

```diff
@@ -0,0 +1,28 @@
+package org.eclipse.swt.internal;
+
+import static org.junit.Assume.assumeTrue;
+
+import java.net.*;
+
+public final class PlatformSpecificExecution {
+	private PlatformSpecificExecution()  {
+	}
+
+	public static void assumeIsFittingPlatform() {
+		assumeTrue("test is specific for Windows", isFittingOS());
+		assumeTrue("architecture of platform does not match", isFittingArchitecture());
+	}
+
+	private static boolean isFittingOS() {
+		return System.getProperty("os.name").toLowerCase().startsWith("win");
```

Comment:
I somehow remmber we already have a way in SWT to determine the platform.

Reply:
org.eclipse.swt.tests.junit.SwtTestUtil.isWindows

Reply:
Yes, there are at least two ways to determine the platform, but unfortunately none of them is applicable here:
1. The platform for which the currently accessed fragment as stored in `Platform.PLATFORM`. At usual runtime, you have the fragment matching your OS loaded (as defined by the platform filter of the fragment), so you can use this to determine the platform you are running. When executing these tests, however, you will always access the `Platform` class from the fragment that you are currently processing, so `Platform.PLATFORM` will always be "win32" in this case, even if you are running on Linux or MacOS.
2. The `SwtTestUtil` (as referred to by Ed). But they are placed in the downstream test plug-in and not in the SWT plug-in itself.

But since I have moved this util class to the same package as the `Library` class to retrieve the system architecture from the `Library` class, I can now also retrieve the OS from that class (which does the same as the `SwtTestUtil`). I have adapted this method accordingly in 91a97ca.